src/snap.c

```diff
@@ -196,10 +279,21 @@ snap_grow_to_next_edge(struct view *view, enum view_edge direction,
 	struct border next_edges;
 	edges_initialize(&next_edges);
 
+	/* Invalidate last-snapped-edge cache if view or direction changes */
+	validate_snap_cache(view, direction);
+
 	/* Limit motion to any intervening edge of other views on this output */
 	edges_find_neighbors(&next_edges,
 		view, *geo, output, check_edge,
 		/* use_pending */ true, /* ignore_hidden */ false);
+
+	/*
+	 * Cache the snapped edge so that two successive snapping attempts
+	 * don't get stuck continually trying to hit the same edge if the
+	 * client fails to honor a geometry change.
+	 */
```

Comment:
Maybe something like `[..] if the client doesn't want to honor a geometry change due to client side size constrains.` ?